### ISSUE TITLE
Can't plot_cells to visualize individual genes

### ISSUE THREAD

**Issue by camilafarias112** [cole-trapnell-lab/monocle3#161]
Hi!
I am trying to `plot_cells` to visualize individual genes but I am getting the following error that I haven't seen here in your Github:

```
Error in `$<-.data.frame`(`*tmp*`, "feature_label", value = character(0)) : 
  replacement has 0 rows, data has 2212
```

This is how I did:
1) I created my _cds_ `new_cell_data_set`, which is:
```
class: cell_data_set 
dim: 12473 2212 
metadata(1): cds_version
assays(1): counts
rownames(12473): X0610007P14Rik X0610030E20Rik ... Zxdc mt.Tt
rowData names(1): genenames
colnames(2212): bcGBQW_1 bcFAOT_1 ... bcELUQ_2 bcELOL_2
colData names(5): cell.names orig.ident celltype.groups celltype Size_Factor
reducedDimNames(0):
spikeNames(0):
```
```
> head(gene_annotation$genenames)
[1] X0610007P14Rik X0610030E20Rik X1110004F10Rik X1110008F13Rik X1110028F18Rik X1110037F02Rik
12473 Levels: A030012G06Rik A130010J15Rik A130023I24Rik A130071D04Rik A230004M16Rik A230028O05Rik A230046K03Rik ... Zzz3
```

2) And then:
```
cds <- preprocess_cds(cds, num_dim = 20)
cds <- reduce_dimension(cds)
cds <- cluster_cells(cds)
cds <- learn_graph(cds)
```
With this everything runs smoothly and my _cds_ becomes:
```
class: cell_data_set 
dim: 12473 2212 
metadata(1): cds_version
assays(1): counts
rownames(12473): X0610007P14Rik X0610030E20Rik ... Zxdc mt.Tt
rowData names(1): genenames
colnames(2212): bcGBQW_1 bcFAOT_1 ... bcELUQ_2 bcELOL_2
colData names(5): cell.names orig.ident celltype.groups celltype Size_Factor
reducedDimNames(2): PCA UMAP
spikeNames(0):
```

3) This is where I get the error message:
```
plot_cells(cds,
           genes = c("Gzmb"),
           label_cell_groups=FALSE,
           show_trajectory_graph=FALSE)
```

What would be this "replacement"? Seems like I am calling my gene from a different input to plot?

**Comment by hpliner** [comment]
Thanks for the report. Can you please post the traceback() from the error?

**Comment by friedue** [comment]
Hi,

I get the same error, following virtually the same workflow as described above.

```
> plot_cells(cds, genes=c( "naga" ),show_trajectory_graph=FALSE,label_cell_groups=FALSE,label_leaves=FALSE)
Error in `$<-.data.frame`(`*tmp*`, "feature_label", value = character(0)) : 
  replacement has 0 rows, data has 3743
> traceback()
4: stop(sprintf(ngettext(N, "replacement has %d row, data has %d", 
       "replacement has %d rows, data has %d"), N, nrows), domain = NA)
3: `$<-.data.frame`(`*tmp*`, "feature_label", value = character(0)) at plotting.R#554
2: `$<-`(`*tmp*`, "feature_label", value = character(0)) at plotting.R#554
1: plot_cells(cds, genes = c("naga"), show_trajectory_graph = FALSE, 
       label_cell_groups = FALSE, label_leaves = FALSE)

> sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server release 6.3 (Santiago)

Matrix products: default
BLAS/LAPACK: /pbtech_mounts/homes022/frd2007/miniconda3/envs/monocle/lib/libopenblasp-r0.3.6.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] shiny_1.3.2                 magrittr_1.5               
 [3] monocle3_0.1.3              testthat_2.2.1             
 [5] SingleCellExperiment_1.7.4  SummarizedExperiment_1.15.6
 [7] DelayedArray_0.11.4         BiocParallel_1.19.0        
 [9] matrixStats_0.54.0          GenomicRanges_1.37.14      
[11] GenomeInfoDb_1.21.1         IRanges_2.19.10            
[13] S4Vectors_0.23.17           Biobase_2.45.0             
[15] BiocGenerics_0.31.5        

loaded via a namespace (and not attached):
  [1] Rtsne_0.15               colorspace_1.4-1         deldir_0.1-23           
  [4] grr_0.9.5                pryr_0.1.4               class_7.3-15            
  [7] rprojroot_1.3-2          XVector_0.25.0           fs_1.3.1                
 [10] rstudioapi_0.10          proxy_0.4-23             listenv_0.7.0           
 [13] furrr_0.1.0              remotes_2.1.0            ggrepel_0.8.1           
 [16] splines_3.6.1            codetools_0.2-16         pscl_1.5.2              
 [19] pkgload_1.0.2            speedglm_0.3-2           jsonlite_1.6            
 [22] RhpcBLASctl_0.18-205     uwot_0.1.3               pheatmap_1.0.12         
 [25] compiler_3.6.1           httr_1.4.1               backports_1.1.4         
 [28] assertthat_0.2.1         Matrix_1.2-17            lazyeval_0.2.2          
 [31] limma_3.41.15            cli_1.1.0                later_0.8.0             
 [34] htmltools_0.3.6          prettyunits_1.0.2        tools_3.6.1             
 [37] igraph_1.2.4.1           coda_0.19-3              gtable_0.3.0            
 [40] glue_1.3.1               GenomeInfoDbData_1.2.1   RANN_2.6.1              
 [43] reshape2_1.4.3           dplyr_0.8.3              gmodels_2.18.1          
 [46] Rcpp_1.0.2               slam_0.1-45              spdep_1.1-2             
 [49] gdata_2.18.0             nlme_3.1-141             DelayedMatrixStats_1.7.1
 [52] lmtest_0.9-37            stringr_1.4.0            globals_0.12.4          
 [55] ps_1.3.0                 mime_0.7                 irlba_2.3.3             
 [58] gtools_3.8.1             devtools_2.1.0           future_1.14.0           
 [61] LearnBayes_2.15.1        zlibbioc_1.31.0          MASS_7.3-51.4           
 [64] zoo_1.8-6                scales_1.0.0             promises_1.0.1          
 [67] expm_0.999-4             RColorBrewer_1.1-2       gridExtra_2.3           
 [70] memoise_1.1.0            reticulate_1.13          pbapply_1.4-1           
 [73] ggplot2_3.2.1            Matrix.utils_0.9.7       stringi_1.4.3           
 [76] desc_1.2.0               e1071_1.7-2              boot_1.3-23             
 [79] pkgbuild_1.0.4           spData_0.3.0             rlang_0.4.0             
 [82] pkgconfig_2.0.2          bitops_1.0-6             rsample_0.0.5           
 [85] lattice_0.20-38          purrr_0.3.2              sf_0.7-6                
 [88] labeling_0.3             htmlwidgets_1.3          processx_3.4.1          
 [91] tidyselect_0.2.5         plyr_1.8.4               R6_2.4.0                
 [94] generics_0.0.2           DBI_1.0.0                pillar_1.4.2            
 [97] withr_2.1.2              units_0.6-3              sp_1.3-1                
[100] RCurl_1.95-4.12          tibble_2.1.3             crayon_1.3.4            
[103] plotly_4.9.0             viridis_0.5.1            usethis_1.5.1           
[106] grid_3.6.1               data.table_1.12.2        callr_3.3.1             
[109] digest_0.6.20            classInt_0.3-1           pbmcapply_1.5.0         
[112] xtable_1.8-4             tidyr_0.8.3              httpuv_1.5.1            
[115] RcppParallel_4.4.3       munsell_0.5.0            viridisLite_0.3.0       
[118] sessioninfo_1.1.1  
```

**Comment by hpliner** [comment]
Hello, thanks for updating the report. Looks like this was a bug related to not having a "gene_short_name" column in your feature data. This is now fixed in the develop version and the fix will be released as part of v0.1.4. You can update to the develop branch by running ```devtools::install_github(cole-trapnell-lab/monocle3, ref = "develop")```

**Comment by friedue** [comment]
that was my suspicion; thanks for fixing it!
